tool/src/main/java/org/wildfly/security/tool/MaskCommand.java

```diff
@@ -162,4 +169,33 @@ public void help() {
                 "",
                 true);
     }
+
+    /**
+     * Check for FIPS (mask password command can't be used in FIPS mode).
+     *
+     * @return {@code true} if running in FIPS mode, {@code false} otherwise
+     */
+    private static boolean isFIPS() {
+        if (new SecureRandom().getProvider().getName().toLowerCase().contains("fips")) {
+            return true;
+        } else {
+            try {
+                final Class<?> providerClazz = MaskCommand.class.getClassLoader().loadClass("com.sun.net.ssl.internal.ssl.Provider");
+                final Method isFipsMethod = providerClazz.getMethod("isFIPS", new Class[0]);
+
+                try {
+                    Object isFips = isFipsMethod.invoke(null, new Object[0]);
```

Comment:
This access can result in the following warning that can't be avoided:

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.wildfly.security.tool.MaskCommand (file:/home/diana/wildfly-elytron/tool/target/wildfly-elytron-tool-shaded-1.18.2.CR1-SNAPSHOT.jar) to method com.sun.net.ssl.internal.ssl.Provider.isFIPS()
WARNING: Please consider reporting this to the maintainers of org.wildfly.security.tool.MaskCommand
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```
If we want to avoid this warning 1 option is to not check FIPS mode and print information that mask password is not allowed in FIPS mode every time and proceed with computation for all cases. Enforced FIPS mode would result in the exception but the information why would be printed out as well. Another option is to make the check weaker and only use `new SecureRandom().getProvider().getName().toLowerCase().contains("fips")`  as a check.